rest/runtime/src/test/java/io/syndesis/runtime/connector/CustomConnectorITCase.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.runtime.connector;
+
+import java.util.List;
+
+import io.syndesis.connector.generator.ConnectorGenerator;
+import io.syndesis.model.connection.Connector;
+import io.syndesis.model.connection.ConnectorTemplate;
+import io.syndesis.model.connection.ConnectorSettings;
+import io.syndesis.rest.v1.operations.Violation;
+import io.syndesis.runtime.BaseITCase;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ContextConfiguration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ContextConfiguration
+public class CustomConnectorITCase extends BaseITCase {
+
+    private final ConnectorTemplate template;
+
+    public static class ConnectorTemplateResultList {
+        public List<ConnectorTemplate> items;
+
+        public int totalCount;
```

Comment:
could be later verified, too.

Reply:
I would not, as it gets data from `deployment.json`, so the number of templates will vary over time and the test would need to be changed. The test checks if the the template it sets up is present, presumably we are not testing DataManager implementation here...